### ISSUE TITLE
Python 3.7 compatibility: Homebrew now installs 3.7 by default

### ISSUE THREAD

**Issue by bgyori** [sorgerlab/indra#624]
It turns out, Homebrew has already moved to Python 3.7 by default i.e. when doing
```brew install python```, users currently end up with 3.7.0. Based on some preliminary tests, there are at least the following issues that make it difficult to support 3.7:
- Cython 0.23.5 doesn't work with 3.7 and so the version constraint would need to be removed from our setup
- But PyJnius (or at least jnius-indra) doesn't seem to work with the latest Cython version (and so changing the Cython version is a problem even with earlier versions of Python)
- Travis doesn't fully support 3.7 yet, one has to use a workaround like the one below for testing
```
matrix:
  include:
    - python: 3.7
      dist: xenial
      sudo: true
```
- kappy==4.0.0rc1 doesn't have a 3.7 install option available on PyPI

**Comment by bgyori** [comment]
After much experimentation with pyjnius and Cython versions, #625 solves the first two issues listed above. I also packaged Kappy for Python 3.7 on Mac and Linux and sent it to @pirbo to publish on PyPI.

**Comment by bgyori** [comment]
The Travis tests reveal the following errors from third-party packages in Python 3.7:
1. There is a generator-related error in Pygraphviz (https://github.com/pygraphviz/pygraphviz/issues/168)
2. Tweepy can't be imported (https://github.com/tweepy/tweepy/issues/1063)

Both of these dependencies are non-critical, the latest Github masters of both projects have fixed these issues, and will soon be released on PyPI.

**Comment by bgyori** [comment]
Pygraphviz 1.5 was released which works with Python 3.7. All tests on Travis are passing (on a branch). For now, I'm not turning on testing for 3.7 by default on Travis but we can do that at some later point.
